moquette-0.17/broker/src/main/java/io/moquette/broker/subscriptions/CTrie.java

```diff
@@ -159,33 +180,34 @@ public void removeFromTree(Topic topic, String clientID) {
 
     private Action remove(String clientId, Topic topic, INode inode, INode iParent) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 197-211 into a separate method.